views/search_view/views.py

```diff
@@ -0,0 +1,81 @@
+from django.shortcuts import render
+from django.views import View
+
+from core.user_controller.UserController import UserController
+from core.course_controller.CourseController import CourseController
+from ta_scheduler.models import Semester
+
+
+class SearchView(View):
+    def get(self, request, type: str):
+        """
+        GET request is used to initialize the page:
+        - If type is "course", all courses are displayed.
+        - If type is "user", display prompt information or empty content.
+        """
+        context = {
+            'full_name': f"{request.user.first_name} {request.user.last_name}",
+            'isAdmin': request.user.role == 'Admin',
+            "type": type,
+            "search_results": [],
+        }
+
+        if type == "course":
+            context["semesters"] = Semester.objects.all()
+            for semester in Semester.objects.all():
+                semester_courses = CourseController.search_courses("", semester.semester_name)
```

Comment:
I think we should refactor the course controller later so that we don't have to do this.
But this is a good solution for now.